src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1083,6 +1088,89 @@ class spell_warr_victory_rush : public SpellScript
     }
 };
 
+// 228920 - Ravager
+class spell_warr_ravager : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_RAVAGER_SUMMON });
+    }
+
+    void SummonRavager(SpellEffIndex /*effIndex*/) const
+    {
+        GetCaster()->CastSpell(*GetHitDest(), SPELL_WARRIOR_RAVAGER_SUMMON, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell()
+        });
+    }
+
+    void Register() override
+    {
+        OnEffectHit += SpellEffectFn(spell_warr_ravager::SummonRavager, EFFECT_1, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 228920 - Ravager - Periodic AuraScript
+class spell_warr_ravager_aurascript : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_RAVAGER_PERIODIC_DAMAGE })
+            && ValidateSpellEffect({ { SPELL_WARRIOR_RAVAGER_SUMMON, EFFECT_0 } });
+    }
+
+    std::vector<Unit*> FindRavagerSummons(Unit const* owner) const
+    {
+        std::vector<Unit*> summons;
+        int32 const summonedCreatureEntry = sSpellMgr->GetSpellInfo(SPELL_WARRIOR_RAVAGER_SUMMON, GetCastDifficulty())->GetEffect(EFFECT_0).MiscValue;
```

Comment:
why int, a valid creature is never under 1, you can static cast this instead and keep the type as unsigned if you want to avoid the warning about type change due to the getter type